### PR TITLE
Migrate to realm DB + notification changes

### PR DESCRIPTION
**Pull request by vycius**
* Realm db;
* Include text to notifications; 
screenshot_20161118-200407

@justasm Could you upload notification icon ?

### CODE REVIEW

File: app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt [hunk 1]
```diff
@@ -47,68 +54,87 @@ class GeofenceIntentService : IntentService(GeofenceIntentService.TAG) {
...
-        // Send notification and log the transition details.
-        sendNotification()
-        Timber.i(geofenceTransitionDetails)
+        val triggeredStory = getTriggeredStory(triggeringGeofencesIdsList)
+
+        sendNotification(triggeredStory)
+    }
+
+    private fun getTriggeredStory(storyIds: List<String>): Story {
+        return StoryDb.getByIds(storyIds).first()
```
Comment (justasm) [inline app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt]: Would it not make sense to just query with one id?

File: app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt [hunk 2]
```diff
@@ -47,68 +54,87 @@ class GeofenceIntentService : IntentService(GeofenceIntentService.TAG) {
...
+                }
+            }
+
+            override fun onPrepareLoad(placeHolderDrawable: Drawable?) {
+            }
+
+        }
+
+        Handler(mainLooper).post({
+            Picasso.with(this)
```
Comment (justasm) [inline app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt:136]: Interesting, is Picasso not thread-safe?

**Review by justasm — APPROVED** [review]
Haven't reviewed Realm docs.. but it seems to work great 😄 Any idea how much it bloats the APK?

File: app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt (hunk 2 above)
Comment (vycius) [reply app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt:136]: I thought that it is. Although Picasso has check from which thread it was called.

``` java
  static void checkMain() {
    if (!isMain()) {
      throw new IllegalStateException("Method call should happen from the main thread.");
    }
  }
```

When starting from another thread:

```
java.lang.IllegalStateException: Method call should happen from the main thread.
                                                                              at com.squareup.picasso.Utils.checkMain(Utils.java:136)
                                                                              at com.squareup.picasso.RequestCreator.into(RequestCreator.java:496)
                                                                              at lt.gyvosistorijos.service.GeofenceIntentService.sendNotification(GeofenceIntentService.kt:134)
                                                                              at lt.gyvosistorijos.service.GeofenceIntentService.onHandleIntent(GeofenceIntentService.kt:69)
                                                                              at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:68)
                                                                              at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                              at android.os.Looper.loop(Looper.java:154)
                                                                              at android.os.HandlerThread.run(HandlerThread.java:61)
```

File: app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt (hunk 1 above)
Comment (vycius) [reply app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt]: Yes. In this implementation it make sense.

**Comment by vycius** [comment]
About .apk sizes:
With realm: 20.9 MB
Old implementation: 15.3 MB

File: app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt (hunk 2 above)
Comment (justasm) [reply app/src/main/java/lt/gyvosistorijos/service/GeofenceIntentService.kt:136]: You're right, just checked myself.

**Comment by vycius** [comment]
Final notification look with icon
screenshot_20161119-143037

**Pull request merged**